### PR TITLE
Client multiple phone numbers & primary number (DEV-177)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-177

This PR:
- creates a generic `PhoneNumber` model in `common`
- moves associated phone number types to `common`
- adds a `phone_numbers` field to `ClientProfile`

Follow-up tickets: DEV-751 (fe update & cleanup), DEV-752 (be update & cleanup)

### CODE REVIEW

File: apps/betterangels-backend/clients/schema.py [hunk 1]
```diff
@@ -32,31 +32,74 @@
...
-    model_cls = apps.get_model("clients", model_cls_name)
+    model_cls = apps.get_model(module, model_cls_name)
 
     item_updates_by_id = {item["id"]: item for item in data if item.get("id")}
     items_to_create = [item for item in data if not item.get("id")]
-    items_to_update = model_cls.objects.filter(id__in=item_updates_by_id.keys(), client_profile=client_profile)
+    items_to_update = model_cls.objects.filter(id__in=item_updates_by_id.keys())
     model_cls.objects.exclude(id__in=item_updates_by_id).delete()
 
+    if generic_relation:
```
Comment (mikefeldberg) [inline apps/betterangels-backend/clients/schema.py:83]: not sure if this is overloading the method and we should split this up differently

File: apps/betterangels-backend/clients/schema.py [hunk 2]
```diff
@@ -32,31 +32,74 @@
     UpdateClientProfileInput,
 )
 
-CLIENT_RELATED_CLS_NAME_BY_RELATED_NAME = {
-    "contacts": "ClientContact",
-    "hmis_profiles": "HmisProfile",
-    "household_members": "ClientHouseholdMember",
-    "social_media_profiles": "SocialMediaProfile",
+CLIENT_RELATED_CLS_NAME_BY_RELATED_NAME: dict[str, dict[str, Any]] = {
```
Comment (mikefeldberg) [inline apps/betterangels-backend/clients/schema.py:35]: this dict can be simplified by setting some default values in the mutation. e.g.

```python
    related_cls_map["module"],
    related_cls_map["cls_name"],
    related_cls_map["generic_relation"],
```
-> 
```python
    related_cls_map.get("module", "clients"),
    related_cls_map["cls_name"],
    related_cls_map.get("generic_relation", "False"),
```

File: apps/betterangels-backend/common/models.py [hunk 3]
```diff
@@ -220,6 +221,25 @@ def get_or_create_location(cls, location_data: Dict[str, Any]) -> "Location":
         return location
 
 
+class PhoneNumber(models.Model):
```
Comment (vecchp) [inline apps/betterangels-backend/common/models.py:224]: Probably need to follow the index pattern similar to the attachment model.  We likely want to have an index on `context_type` and `object_id`

File: apps/betterangels-backend/common/models.py [hunk 4]
```diff
@@ -220,6 +221,25 @@ def get_or_create_location(cls, location_data: Dict[str, Any]) -> "Location":
...
+    number = PhoneNumberField(region="US", blank=True, null=True)
+    is_primary = models.BooleanField(default=False)
+
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField()
+    content_object: GenericForeignKey = GenericForeignKey("content_type", "object_id")
```
Comment (vecchp) [inline apps/betterangels-backend/common/models.py:230]: Is this used automatically?

Comment (mikefeldberg) [reply apps/betterangels-backend/common/models.py:230]: not sure I understand the question. Automatically as opposed to needing to be defined/invoked somewhere?

Comment (vecchp) [reply apps/betterangels-backend/common/models.py:230]: Related to previous question, if generic foreign keys create indices then the previous recommendation isn't necessary.

Comment (vecchp) [reply apps/betterangels-backend/common/models.py:230]: See: https://docs.djangoproject.com/en/5.1/ref/contrib/contenttypes/#generic-relations

File: apps/betterangels-backend/common/graphql/types.py [hunk 5]
```diff
@@ -1,11 +1,33 @@
...
+    DjangoPhoneNumber,
+    serialize=lambda v: str(v.national_number),
+    parse_value=lambda v: parse(v, "US"),
+)
+
+
+@strawberry_django.type(PhoneNumber)
+class PhoneNumberType:
+    id: ID
+    number: Optional[PhoneNumberScalar]  # type: ignore
```
Comment (vecchp) [inline apps/betterangels-backend/common/graphql/types.py:21]: What is the type error here, why ignored?

File: apps/betterangels-backend/clients/schema.py (hunk 1 above)
Comment (vecchp) [reply apps/betterangels-backend/clients/schema.py:83]: This function is getting pretty big.

File: apps/betterangels-backend/clients/schema.py (hunk 2 above)
Comment (vecchp) [reply apps/betterangels-backend/clients/schema.py:35]: Isn't there a way to directly get the cls from the relation?

File: apps/betterangels-backend/common/graphql/types.py (hunk 5 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/common/graphql/types.py:21]: this is the same error we're ignoring elsewhere in [ClientProfileBaseType](https://github.com/BetterAngelsLA/monorepo/blob/ccab1e629b0f44831e4d15823e95bb1241878f1a/apps/betterangels-backend/clients/types.py#L162) and [ClientContactBaseType](https://github.com/BetterAngelsLA/monorepo/blob/ccab1e629b0f44831e4d15823e95bb1241878f1a/apps/betterangels-backend/clients/types.py#L186). It's a known strawberry/mypy issue we ran into when we implemented django-phonenumber-field

File: apps/betterangels-backend/common/models.py (hunk 4 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/common/models.py:230]: ya, there's no index created. I updated that

**Review by vecchp — APPROVED** [review]

**Pull request merged**